File: 2015/day-01/main.lua
function process_instructions(instructions)
    local floor = 0
    for i = 1, #instructions do
        local instruction = instructions:sub(i, i)
        if instruction == "(" then
            floor = floor + 1
        elseif instruction == ")" then
            floor = floor - 1
        end
    end
    return floor
end

-- read input from file
local f = io.open("input.txt", "r")
local instructions = f:read("*all")
f:close()

local final_floor = process_instructions(instructions)
print("Santa ended up on floor " .. final_floor)

File: 2015/day-01/Main.java
import java.io.File;
import java.io.FileNotFoundException;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) throws FileNotFoundException {
        // Read the input file
        File inputFile = new File("input.txt");
        Scanner scanner = new Scanner(inputFile);
        String instructions = scanner.nextLine();
        scanner.close();
        
        // Follow the instructions and determine the final floor
        int floor = 0;
        for (int i = 0; i < instructions.length(); i++) {
            if (instructions.charAt(i) == '(') {
                floor++;
            } else if (instructions.charAt(i) == ')') {
                floor--;
            }
        }
        
        // Print the final floor where Santa ended up
        System.out.println("Santa ended up on floor " + floor);
    }
}
